webrender/src/platform/windows/font.rs

```diff
@@ -85,98 +139,88 @@ impl FontContext {
 
         let width = (bounds.right - bounds.left) as u32;
         let height = (bounds.bottom - bounds.top) as u32;
-        let dims = GlyphDimensions {
+        GlyphDimensions {
             left: bounds.left,
             top: -bounds.top,
             width: width,
             height: height,
-        };
-
-        // if empty, then nothing
-        if dims.width == 0 || dims.height == 0 {
-            return (None, None);
         }
+    }
 
-        // if we weren't asked to rasterize, we're done
-        if render_mode.is_none() {
-            return (Some(dims), None);
-        }
+    pub fn get_glyph_dimensions(&self,
+                                font_key: FontKey,
+                                size: Au,
+                                glyph: u32) -> Option<GlyphDimensions> {
+        // Probably have to default to something else here.
+        let render_mode = FontRenderMode::Subpixel;
+        let analysis = self.create_glyph_analysis(font_key, size, glyph, render_mode);
+
+        let texture_type = self.get_dwrite_texture_type(render_mode);
+        Some(self.get_glyph_dimensions_with_analysis(analysis, texture_type))
+    }
 
-        let pixels = analysis.create_alpha_texture(tex_type, bounds);
-        let rgba_pixels = match render_mode.unwrap() {
+    // DWRITE gives us values in RGB. WR expects it in RGBA. Note, CG returns in BGR
+    // TODO: probably should fix that
+    fn convert_to_rgba(&self, pixels: &Vec<u8>, render_mode: FontRenderMode) -> Vec<u8> {
+        match render_mode {
             FontRenderMode::Mono => {
-                let mut rgba_pixels = vec![0; pixels.len() * 4];
+                let mut rgba_pixels = vec![0 as u8; pixels.len() * 4];
                 for i in 0..pixels.len() {
-                    rgba_pixels[i*4+0] = 0xff;
-                    rgba_pixels[i*4+1] = 0xff;
-                    rgba_pixels[i*4+2] = 0xff;
+                    rgba_pixels[i*4+0] = pixels[i];
+                    rgba_pixels[i*4+1] = pixels[i];
+                    rgba_pixels[i*4+2] = pixels[i];
                     rgba_pixels[i*4+3] = pixels[i];
                 }
                 rgba_pixels
             }
             FontRenderMode::Alpha => {
-                let mut rgba_pixels = vec![0; pixels.len()/3 * 4];
-                for i in 0..pixels.len()/3 {
+                let length = pixels.len() / 3;
+                let mut rgba_pixels = vec![0 as u8; length * 4];
+                for i in 0..length {
                     // TODO(vlad): we likely need to do something smarter
-                    let alpha = (pixels[i*3+0] as u32 + pixels[i*3+0] as u32 + pixels[i*3+0] as u32) / 3;
-                    rgba_pixels[i*4+0] = 0xff;
-                    rgba_pixels[i*4+1] = 0xff;
-                    rgba_pixels[i*4+2] = 0xff;
-                    rgba_pixels[i*4+3] = alpha as u8;
+                    // This is what skia does
+                    let alpha :u8 = ((pixels[i*3+0] as u32 +
```

Comment:
spacing `alpha: u8`.  Though you don't need the ': u8' at all, the whole thing gets cast at the end anyway.